Master feature file extension linter

privateOmega opened the pull request:
#27 Hope this code solves the issue. The test conditions were duplicated and adjusted to meet the requirement.

privateOmega commented:
@pietvanzoen please review and merge it

privateOmega commented on .coveralls.yml:
```diff
@@ -0,0 +1 @@
+repo_token: zcfhk98Rjs4AEo4GhiVhUKPRRwMENqh3Q
```
oh yes. will have it removed.

privateOmega replied:
I have no idea how to run coveralls build. I would appreciate it, if you could point me in the right direction.

pietvanzoen replied:
Coveralls does actually run a build. You send the coverage data to coveralls from your CI service. Here's where that happens: https://github.com/usabilla/fnlint/blob/master/scripts/test.sh#L12

privateOmega commented on lib/linter.js:
```diff
@@ -19,7 +19,8 @@ class Linter {
   filePathTest(filePath) {
     const segments = _.compact([].concat(this.pathParser(filePath)));
-    return assertAllSegmentsPassing(segments, this.matcher);
+    if(segments.length)
```
Yes. I encountered a situation where an empty segment would pass, it was undesirable. I am happy to change accordingly, if you have a better idea why it does.

pietvanzoen replied:
That makes sense. If wouldn't mind adding a test that covers this situation that would be grand.

privateOmega replied:
@pietvanzoen I am so sorry for getting back to you this late. I would really appreciate it, if you could add the required test since I don't have a clue on doing it.

marco-feenstra-usabilla commented:
This PR contains two commits containing the coverall secret.
commit e06959b780c149c86ae41f54134c8deeb5d8ad44
Introduces the secret

commit 747810ba2bee3573311b0c95462788505fdd66db
removes the secret.

As such the secret will remain accessible in the Git Versioning system.
Can the PR be updated with an git rebase -i  to remove both commits so the secret will not be stored in the git commit history?

privateOmega commented:
@marco-feenstra-usabilla @pietvanzoen So sorry, I haven't had the time to check up on this PR. I have rebased it and removed the two commits. Hopefully that's good enough to do the merge. If there's anything else, do comment below and I will look into it ASAP.

pietvanzoen commented:
@privateOmega Thanks for the contribution! Very sorry for the delay on getting it merged. All merged now. Version 1.5.0 is the latest.

Cheers!
